blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/astyanax/APreaggregatedMetricsRW.java

```diff
@@ -0,0 +1,133 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.astyanax;
+
+import com.netflix.astyanax.connectionpool.exceptions.ConnectionException;
+import com.netflix.astyanax.model.ColumnFamily;
+import com.rackspacecloud.blueflood.io.AbstractMetricsRW;
+import com.rackspacecloud.blueflood.io.CassandraModel;
+import com.rackspacecloud.blueflood.outputs.formats.MetricData;
+import com.rackspacecloud.blueflood.rollup.Granularity;
+import com.rackspacecloud.blueflood.service.SingleRollupWriteContext;
+import com.rackspacecloud.blueflood.types.*;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class deals with reading/writing metrics to the metrics_preaggregated_* column families
+ * using Astyanax driver
+ */
+public class APreaggregatedMetricsRW extends AbstractMetricsRW {
+
+    /**
+     * Inserts a collection of metrics to the metrics_preaggregated_full column family
+     *
+     * @param metrics
+     * @throws IOException
+     */
+    @Override
+    public void insertMetrics(Collection<IMetric> metrics) throws IOException {
+        insertMetrics(metrics, Granularity.FULL);
+    }
+
+    /**
+     * Inserts a collection of metrics to the correct column family based on
+     * the specified granularity
+     *
+     * @param metrics
+     * @param granularity
+     * @throws IOException
+     */
+    @Override
+    public void insertMetrics(Collection<IMetric> metrics, Granularity granularity) throws IOException {
+        try {
+            AstyanaxWriter.getInstance().insertMetrics(metrics, CassandraModel.getPreaggregatedColumnFamily(granularity));
+        } catch (ConnectionException ex) {
+            throw new IOException(ex);
+        }
+    }
+
+    @Override
+    public void insertRollups(List<SingleRollupWriteContext> writeContexts) throws IOException {
+        try {
+            AstyanaxWriter.getInstance().insertRollups(writeContexts);
+        } catch (ConnectionException ex) {
+            throw new IOException(ex);
+        }
+    }
+
+    /**
+     * Fetches {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} objects for the
+     * specified {@link com.rackspacecloud.blueflood.types.Locator} and
+     * {@link com.rackspacecloud.blueflood.types.Range} from the specified column family
+     *
+     * @param locator
+     * @param range
+     * @param gran
+     * @return
+     */
+    @Override
+    public MetricData getDatapointsForRange(final Locator locator,
+                                            Range range,
+                                            Granularity gran) throws IOException {
+        return AstyanaxReader.getInstance().getDatapointsForRange(locator, range, gran);
```

Comment:
getDatapointsForRange() method is defined in the MetricsRW interface. Both Astyanax and Datastax implements this. The Datastax will throw IOException. The Astayanx may not. Should I catch a generic RuntimeException here and rethrow as IOException ?